src/actions/indexing/indexing-complete-action.php

```diff
@@ -31,7 +31,7 @@ public function __construct( Options_Helper $options ) {
 	 * @return void
 	 */
 	public function complete() {
-		$this->options->set( 'indexation_started', null );
-		$this->options->set( 'indexing_reason', '' );
+		$this->indexing_helper->set_started( null );
+		$this->indexing_helper->set_reason( '' );
```

Comment:
You might consider adding a method to the indexing_helper: `finish` - which contains this logic.

Reply:
An alternative is adding a `clear_reason`. Which is a logical and clear method. 

I can imagine a scenario that sets a reason and also does other things like removed a notification or that kind of stuff. If you have one method for setting and 'unsetting' the value you have to add additional logic.

Reply:
Done. I created the `finish` function, because this is more consistent with the `start` function that I created in response to your comment below.